app_architecture/ruby/ruby_on_rails.adoc

```diff
@@ -0,0 +1,332 @@
+====
+
+For this guide you will need:
+
+* Basic Ruby/Rails knowledge
+* Locally installed version of Ruby 2.0.0+, Rubygems, Bundler
+* Basic Git knowledge
+* Running instance of OpenShift v3
+
+== Local Workstation Setup
+
+As first make sure that an instance of OpenShift is running and is available. For more info on
+how the get OpenShift up and running check the link:../../getting_started/administrators.html#installation-methods[installation methods]. Also make sure that your link:../../cli_reference/get_started_cli.html[`oc` CLI client is installed] and the command is accessible from your command shell, so you can use it to link:../../cli_reference/get_started_cli.html#basic-setup-and-login[log in] using the email address and password. 
+
+== Write your application
+
+If you are starting your Rails application from scratch, you need to install the Rails gem at first.
+
+----
+$ gem install rails
+Successfully installed rails-4.2.0
+1 gem installed
+----
+
+After you install the Rails gem create a new application:
+
+----
+$ rails new rails-app --database=postgresql
+----
+
+Then move into your new application directory.
+
+----
+$ cd rails-app
+----
+
+In case you already have an application, make sure the `pg` gem is present in your `Gemfile`. If not
+edit your `Gemfile` by adding the gem:
+
+----
+gem 'pg'
+----
+
+To generate a new `Gemfile.lock` with all your dependencies run:
+
+----
+$ bundle install
+----
+
+In addition to using the `postgresql` database with `pg` gem, you'll also need to ensure the `config/databese.yml` is using the `postgresql` adapter.
+
+Make sure your development section of your `config/database.yml` file looks like this:
+
+----
+# PostgreSQL. Versions 8.2 and up are supported.
+#
+# Install the pg driver:
+#   gem install pg
+# On OS X with Homebrew:
+#   gem install pg -- --with-pg-config=/usr/local/bin/pg_config
+# On OS X with MacPorts:
+#   gem install pg -- --with-pg-config=/opt/local/lib/postgresql84/bin/pg_config
+# On Windows:
+#   gem install pg
+#       Choose the win32 build.
+#       Install PostgreSQL and put its /bin directory on your path.
+#
+# Configure Using Gemfile
+# gem 'pg'
+#
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see rails configuration guide
+  # http://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: 5
+----
+
+[WARNING]
+====
+*ToBeDone*:
+We in OpenShift need a way how to complete the database.yml based on the detected available environment
+variables, Gemfile gems and set adapter in `config/database.yml`. For example how the final file should look like, check out our link:https://github.com/openshift/rails-ex[Ruby on Rails example application] link:https://github.com/openshift/rails-ex/blob/master/config/database.yml[config/database.yml].
```

Comment:
for now just document it with hardcoded values like we have in our sample repo, since you are having them define the service and environment variables explicitly later, you know it will be correct and i don't want this doc to be held up on us fixing up the ruby image to generate a database.yml for you.